src/web3-contracts.js

```diff
@@ -295,7 +305,17 @@ export function useConvertTokenToAnj(selectedToken) {
       // If the user has selected ANT, we can directly
       // approve and call the wrapper using ANT's approveAndCall
       if (selectedToken === 'ANT') {
-        return tokenContract.approveAndCall(wrapperAddress, amount, '0x00', {
+        const encodedActivation = ethers.utils.hexlify(bigNum(1)).split('0x')[1]
+        const encodedMinTokens = ethers.utils
+          .hexlify(underestimatedAnj)
+          .split('0x')[1]
+        const encodedMinEth = ethers.utils.hexlify(estimatedEth).split('0x')[1]
+        const encodedDeadline = ethers.utils
+          .hexlify(twoHourExpiry)
+          .split('0x')[1]
+
+        const data = `0x${encodedActivation}${encodedMinTokens}${encodedMinEth}${encodedDeadline}`
```

Comment:
I'll need some more context here... I'm pretty sure this is not the correct way to concatenate the strings for the data bytes array. Is there a better / cleaner way?

Reply:
So the way I would do this:

```js
const encodedActivation = (activate ? '1' : '0').padStart(64, '0')
const encodedMinTokens = minTokens.toHexString().slice(2).padStart(64, '0')
...

const data = `0x${encodedActivation}${...}`
```

`padStart()` is quite useful for encoding these types of bytes strings :).